my_Enumerable.rb

```diff
@@ -0,0 +1,153 @@
+      end
+      return self
+    end
+
+    def my_select
+      return to_enum(:double) unless block_given?
+      conditionMet = []
+      self.my_each do |item|
+        if yield(item)
+          conditionMet << item
+        end
+      end
+      conditionMet
+    end
+
+    def my_all?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          unless yield(item)
+            return false
+          end
+        else
+          if pattern
+            unless pattern === (item)
+              return false
+            end
+          else
+            if item == nil||item == false
+              return false
+            end
+          end
+        end
+      end
+      return true
+    end
+
+    def my_any?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return true
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return true
+            end
+          else
+            if item == nil||item == false
+              return true
+            end
+          end
+        end
+      end
+      return false
+    end
+
+    def my_none?(pattern = nil)
+      self.my_each do |item|
+        if block_given?
+          if yield(item)
+            return false
+          end
+        else
+          if pattern
+            if pattern === (item)
+              return false
+            end
+          else
+            if item == nil||item == false
+              return false
+            end
+          end
+        end
+      end
+      return true
+    end
+
+    def my_count(value = nil)
+      counter = 0
+      self.my_each do |item|
+        if value == nil &&  !block_given?
+          counter += 1
+        else
+          if block_given?
+            if yield(item)
+              counter += 1
+            end
+          else
+            if item == value
+              counter += 1
+            end
+          end
+        end
+      end
+      return counter
+    end
+
+    def my_map(proc = nil)
+      return to_enum(:double) unless block_given? || !proc
+      newArray = []
+      self.my_each do |item|
+          if proc
+            newArray << proc.call(item)
+          else
+            newArray << yield(item)
+          end
+      end
+      return newArray
+    end
+
+    def my_inject(initial = 0, operator = nil)
+      if initial.class == Symbol
+        operator = initial
+        initial = 0
+      end
+      finalValue = 0
+      arrayToCalc = []
+      arrayToCalc = self.to_a
+      finalValue = arrayToCalc[initial]
+      arrayToCalc.my_each_with_index do |item, i|
+        if i == initial
+          next
+        end
+        if operator!=nil && !block_given?
+          finalValue = finalValue.public_send(operator.to_sym, item)
+        else
+          if block_given?
+            finalValue = yield(finalValue, item)
+          end
+        end
+      end
+      return finalValue
+    end
```

Comment:
### 4. problems with inject, if we add a number of the argument, the accumulator should be initiated with that argument

```
irb(main):195:0> arr = [1,2,3]
=> [1, 2, 3]
irb(main):196:0> arr.inject(2) { |n, z| n + z }
=> 8
irb(main):197:0> arr.my_inject(2) { |n, z| n + z }
=> 6
irb(main):198:0> (1..3).inject(2) { |prod, n| prod * n}
=> 12
irb(main):199:0> (1..3).my_inject(2) { |prod, n| prod * n}
=> 6
```

---------------------------------

### 5. a symbol should be able to pass through and be applied as a operation through all the objects to which we apply the method


```
irb(main):200:0> (1..3).inject(2, :+) 
=> 8
irb(main):201:0> (1..3).my_inject(2, :+) 
=> 6
irb(main):202:0> (1..3).inject(2, :*) 
=> 12
irb(main):203:0> (1..3).my_inject(2, :*) 
=> 6
irb(main):204:0> (1..3).inject(5, :*) 
=> 30
```

Reply:
Ohhh... I assumed for inject the initial value was linked to the element selected from the range or array....
Not that the initial value was the actual value the inject starts off as...